domain/port/state/state.go

```diff
@@ -185,28 +185,33 @@ func (st *State) UpdateUnitPorts(
 	unitUUID := unitUUID{UUID: unit}
 
 	return domain.Run(ctx, func(ctx context.Context, tx *sqlair.TX) error {
-		currentOpenedPorts, err := st.getUnitOpenedPorts(ctx, tx, unitUUID)
+		colocatedOpenedPorts, err := st.getColocatedOpenedPorts(ctx, tx, unitUUID)
 		if err != nil {
-			return errors.Annotatef(err, "getting opened ports for unit %q", unit)
+			return errors.Annotatef(err, "getting opened ports co-located with unit %q", unit)
 		}
 
 		allInputPortRanges := append(openPorts.UniquePortRanges(), closePorts.UniquePortRanges()...)
-		err = verifyNoPortRangeConflicts(currentOpenedPorts, allInputPortRanges)
+		err = verifyNoPortRangeConflicts(colocatedOpenedPorts, allInputPortRanges)
```

Comment:
NOTE: In Mongo, we actually disallow equal port ranges being opened on co-located units, despite the fact we allow them on different endpoints of the same unit

So this behaviour here is slightly different. We will now allow equal opened port ranges on co-located units.

This is fine because downstream when we open port ranges we de-dup across endpoints. We will just do that against co-located units as well